### PR TITLE
Upgrade ember cli to ember-cli 2.18.2

### PR DESCRIPTION
**Pull request by patocallaghan**
I was starting the process of dropping Bower but noticed I needed to do this work first. I wasn't sure whether to drop Bower in this PR but decided to put that in a follow-up one. Happy to add it in here if you want?

* Upgrade addon to ember-cli 2.18.2
* Migrates to new module syntax
* Adds package-lock.json
* Disables `eslint ember/closure-action` rule for `sendAction` code. Temporary until it's decided how to refactor.
* Fixed a subtle timing issue where `this.popup().remove()` is called but the component is destroying
* Fixed some deprecation warnings

### CODE REVIEW

**Comment by patocallaghan** [comment]
Looking into the failing tests

File: addon/mixins/col-pick.js [hunk 1]
```diff
@@ -103,7 +108,9 @@ export default Ember.Mixin.create( {
 
   _tearDownColpick: function() {
     if (this._colpick) {
-      this.popup().remove();
+      if (!this.isDestroying) {
```
Comment (patocallaghan) [inline addon/mixins/col-pick.js]: image

File: addon/mixins/col-pick.js [hunk 2]
```diff
@@ -54,18 +58,19 @@ export default Ember.Mixin.create( {
...
           this.set('previewValue', hex);
 
           if (this._isValidPreviewValue()) {
             this.set('value', hex);
           }
         }),
-        onHide: Ember.run.bind(this, function(){
+        onHide: bind(this, function(){
+          // eslint-disable-next-line ember/closure-actions
           this.sendAction('onHide');
```
Comment (patocallaghan) [inline addon/mixins/col-pick.js]: Disabling this `ember:recommended` in this case until it can be refactored.

File: package.json [hunk 3]
```diff
@@ -2,45 +2,49 @@
...
-    "ember-cli-qunit": "^1.4.0",
+    "broccoli-asset-rev": "^2.4.5",
+    "ember-cli": "~2.18.2",
+    "ember-cli-dependency-checker": "^2.0.0",
+    "ember-cli-eslint": "^4.2.1",
+    "ember-cli-htmlbars": "^2.0.1",
+    "ember-cli-htmlbars-inline-precompile": "^1.0.0",
+    "ember-cli-inject-live-reload": "^1.4.1",
+    "ember-cli-qunit": "^4.1.1",
     "ember-cli-release": "0.2.8",
```
Comment (patocallaghan) [inline package.json]: I wasn't sure whether to remove `ember-cli-release` or not? This isn't in the latest ember-cli package.json but unsure whether it's used in this addon specifically.

**Comment by patocallaghan** [comment]
@lukemelia apologies for the massive diff but I've tried to break things down by commit. Any thoughts?

File: package.json (hunk 3 above)
Comment (lukemelia) [reply package.json]: Please do keep ember-cli-release. I like it for releasing new versions of addons.

**Comment by lukemelia** [comment]
Thanks @patocallaghan. I'll try to give it a review tonight.

**Comment by patocallaghan** [comment]
Much appreciated 🙌

**Comment by lukemelia** [comment]
Thanks @patocallaghan. I'll wait on your followup dropping bower before I cut the next release.

**Pull request merged**